src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -222,18 +214,18 @@ public SymbolTreeInfo WithChecksum(Checksum checksum)
             CancellationToken cancellationToken)
         {
             var comparer = GetComparer(ignoreCase);
-            var results = ArrayBuilder<ISymbol>.GetInstance();
             IAssemblySymbol? assemblySymbol = null;
 
+            using var results = TemporaryArray<ISymbol>.Empty;
             foreach (var node in FindNodeIndices(name, comparer))
             {
                 cancellationToken.ThrowIfCancellationRequested();
                 assemblySymbol ??= await lazyAssembly.GetValueAsync(cancellationToken).ConfigureAwait(false);
 
-                Bind(node, assemblySymbol.GlobalNamespace, results, cancellationToken);
+                Bind(node, assemblySymbol.GlobalNamespace, ref Unsafe.AsRef(in results), cancellationToken);
```

Comment:
def do not like this.  why can we not just pass this by ref?

it's also unclear to me what semantics this has for the dispose in this method.

Reply:
> why can we not just pass this by ref?

Cannot take a `ref` to a `using` variable.

> it's also unclear to me what semantics this has for the dispose in this method.

It's extremely clear because both `ref` and `in` are specified. This is a reference to the same storage location as the using variable.

Reply:
> Cannot take a ref to a using variable.


bleaghhhhhhhhhhhh....

The 'Unsafe' there really stands out like a sore thumb.  Is there any way we can have some sort of method on this like `ref result.PassAsRef()` to be a simpler and clearer pattern for everyone to follow?  It would still do this, just under the covers so it doesn't seem like we're explicitly doing unsafe awfulness at every initial pass site?

Reply:
Aside: would it make sense to have an analyzer that forced all passing of this type ot be either by 'ref' or by 'in'?  i'm a bit worried about code forgetting to do that, and then randomly working or not depending on if it's the arraybuilder that values are being added to, or the inline-items.   Forcing ref/in would prevent htat.

Reply:
> would it make sense to have an analyzer that forced all passing of this type ot be either by 'ref' or by 'in'?

This type is not copyable with analyzer enforcement. Attempting to pass by value will fail the build.

Reply:
> Is there any way we can have some sort of method on this like `ref result.PassAsRef()` to be a simpler and clearer pattern for everyone to follow?

Unfortunately it's not possible to distinguish between the cases we want (referencing a `using` variable) and cases where we really don't want to write to the value (e.g. referencing an `in` parameter).

Reply:
awesome.  i didn't know we had that, but that makes me feel *much* better here as i really care most about people bein able to properly use these types wih confidence.